File: Regression/EP_Web/Links/01_customer_care.rb
#Remove All instances of Current Page from spreadsheet
require 'rubygems'
require 'watir-webdriver'
require "win32ole"

excel= WIN32OLE::new("excel.Application")
workbook=excel.Workbooks.Open("C:\\Scripts\\EP_Web\\Links\\Links.xlsx")
worksheet = workbook.worksheets(1) 
browser = Watir::Browser.new :firefox

rows = 2
while    
rows <= 19

customer=worksheet.cells(rows,"A").value
state=worksheet.cells(rows,"B").value
phone=worksheet.cells(rows,"C").value
email=worksheet.cells(rows,"D").value

browser.goto 'http://www.pt.energypluscompany.com/care/care.php'

browser.radio(:value => customer).click
browser.select_list(:id, 'selectState').select state

if
browser.text.include? phone
print "Phone Passed, "
else
print "Phone Failed, "
end

if 
browser.text.include? email
print "Email Passed", "\n" 
else
print "Email Failed", "\n"
end
 
rows=rows+1
end


browser.close

$end
workbook.Save
workbook.Close


sleep 2

File: Regression/EP_Web/Links/01_customer_care.py
from selenium import webdriver
from selenium.webdriver.common.keys import Keys
from selenium.webdriver.support.select import Select
import time
import xlrd
import os

if os.environ.get('USE_PHANTOM'):
    print("using PhantomJS")

    driver = webdriver.PhantomJS()
    driver.implicitly_wait(5)
    driver.set_window_size(1120, 550)
else:
    print("using Firefox")
    driver = webdriver.Firefox()


driver.get("http://www.pt.energypluscompany.com/care/care.php")

workbook = xlrd.open_workbook("./links.xlsx")
worksheet = workbook.sheet_by_name('Customer Care')

for current_row in range(1,worksheet.nrows):
    customer = worksheet.row(current_row)[0].value
    state = worksheet.row(current_row)[1].value
    phone = worksheet.row(current_row)[2].value
    email = worksheet.row(current_row)[3].value

# radio button existing customer
    elem = driver.find_element_by_xpath(customer).click()
    time.sleep(1)
    
    elem = driver.find_element_by_id("selectState")
    for option in elem.find_elements_by_tag_name('option'):
        if option.text == state:
            option.click()
    elem.click()
    time.sleep(3)

    if  phone in driver.page_source:
        print (state, "Current Customer Phone - Passed")
    else:
        print (state, "Current Customer Phone - Failed")

    if  email in driver.page_source:
        print (state, "Current Customer Email - Passed")
    else:
        print (state, "Current Customer Email - Failed")

driver.close()